src/messenger/messenger.cpp

```diff
@@ -179,9 +182,7 @@ class MessengerPolledSockets {
     if (!writer_.items_.empty()) {
       std::scoped_lock lock(mutex_);
```

Comment:
There could potentially be a race on the check of empty.

Reply:
Yeah, stale reads are OK, it will get caught by the next poll. I will add a comment.

Reply:
Comment added.